pkg/linter/lintRuleSyntacticRedundantCommon.go

```diff
@@ -0,0 +1,229 @@
+		}
+		filterLayer := config.GetFilterTrafficResourceOfKind(filterLayerName)
+		rules, err := filterLayer.GetRules()
+		if err != nil {
+			return nil, err
+		}
+		tableToRules, tableToAtomicBlocks := getTableOrientedStructs(rules)
+		// iterates over tables, in each table iterates over rules and finds those that are redundant (shadowed/implied)
+		for tableIndex, rules := range tableToRules {
+			tableAtomicBlocks := tableToAtomicBlocks[tableIndex]
+			for redundantRuleIndex := range rules {
+				// nacl - redundant if shadowed by over higher priority rules; sg - redundant if implied by "other" rule
+				// in the former case iterates only over relevant slice; in the latter skip the "myself" rule
+				var rulesToIterate []*vpcmodel.RuleOfFilter
+				if filterLayerName == vpcmodel.NaclLayer {
+					rulesToIterate = tableToRules[tableIndex][:redundantRuleIndex]
+				} else { // SG
+					rulesToIterate = tableToRules[tableIndex]
+				}
+				// gathers atomic blocks within rule's src and dst
+				srcBlocks := getAtomicBlocksOfSrcOrDst(tableAtomicBlocks, rules[redundantRuleIndex].SrcCidr)
+				dstBlocks := getAtomicBlocksOfSrcOrDst(tableAtomicBlocks, rules[redundantRuleIndex].DstCidr)
+				// iterates over cartesian product of atomic blocks within rule's src and dst; rule is redundant if all
+				// items in the cartesian product are shadowed/implies
+				ruleIsRedundant := true
+				containRules := map[int]*vpcmodel.RuleOfFilter{}
+				for _, srcAtomicBlock := range srcBlocks {
+					for _, dstAtomicBlock := range dstBlocks {
+						connOfOthers := connection.None()
+						// computes the connection of other/higher priority rules in this atomic point in the 3-dimension space
+						for otherRuleIndex, otherRule := range rulesToIterate {
+							if redundantRuleIndex == otherRuleIndex {
+								continue // relevant only to SG
+							}
+							// otherRule contributes to the shadowing/implication?
+							// namely, it contains src, dst and has a relevant connection?
+							if rules[redundantRuleIndex].IsIngress == otherRule.IsIngress &&
+								srcAtomicBlock.ContainedIn(otherRule.SrcCidr) &&
+								dstAtomicBlock.ContainedIn(otherRule.DstCidr) &&
+								!rules[redundantRuleIndex].Conn.Intersect(otherRule.Conn).IsEmpty() {
+								connOfOthers = connOfOthers.Union(otherRule.Conn)
+								if _, ok := containRules[otherRuleIndex]; !ok {
+									containRules[otherRuleIndex] = otherRule
+								}
+							}
+						}
+						// is <src, dst> shadowed/implied by other rules?
+						if !rules[redundantRuleIndex].Conn.ContainedIn(connOfOthers) {
+							ruleIsRedundant = false
+							break
+						}
+					}
+				}
+				if ruleIsRedundant {
+					res = append(res, ruleRedundant{vpcResource: config.VPC, rule: *rules[redundantRuleIndex], containRules: containRules})
+				}
+			}
+		}
+	}
+	return res, nil
+}
+
+func getAtomicBlocksOfSrcOrDst(atomicBlocks []*ipblock.IPBlock, srcOrdst *ipblock.IPBlock) []*ipblock.IPBlock {
```

Comment:
rename to getAtomicBlocksOfBlock(allAtomicBlocks []*ipblock.IPBlock, block*ipblock.IPBlock)